packages/ckeditor5-autoformat/src/autoformat.js

```diff
@@ -64,6 +65,13 @@ export default class Autoformat extends Plugin {
 		if ( commands.get( 'todoList' ) ) {
 			blockAutoformatEditing( this.editor, this, /^\[\s?\]\s$/, 'todoList' );
 		}
+
+		if ( commands.get( 'checkTodoList' ) ) {
+			blockAutoformatEditing( this.editor, this, /^\[x\]\s$/, () => {
```

Comment:
The regexp should also allow for spaces between x and spaces. I believe it's fairly common to use `[ x ]` too.

Be sure to add TC in unit tests.